index.js

```diff
@@ -69,16 +72,40 @@ module.exports = function (options = {}) {
 				name += `/${parts.shift()}`;
 			}
 
-			if (parts.length > 0) return;
+			const entry = parts.join('/') || '.';
+
+			let pkg;
+			let dir;
 
 			let search_dir = importer;
 			while (search_dir !== (search_dir = path.dirname(search_dir))) {
-				const dir = path.join(search_dir, 'node_modules', name);
+				dir = path.join(search_dir, 'node_modules', name);
 				const file = path.join(dir, 'package.json');
 				if (fs.existsSync(file)) {
-					const pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
-					if (pkg.svelte) {
-						return path.resolve(dir, pkg.svelte);
+					pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
+					break;
+				}
+			}
+
+			if (pkg) {
```

Comment:
Just a matter of style, but you could make this `if (!pkg) return;`. The whole `if` block doesn't fit on one screen so it's hard to see at a glance if there's anything below it, but if you invert the condition it's easy to tell what happens